### PR TITLE
Feat/add copy clipboard

### PR DESCRIPTION
**Pull request by maya-fisher**

### CODE REVIEW

**Review by yarinBenisty — APPROVED** [review]
All good!

File: src/components/buttons/CopyToClipboardButton.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export default {
+  name: "CopyToClipboard",
+  props: ["icon"],
+  computed: {
+    ...mapGetters(["chosenFiles", "currentFolder"]),
+  },
+  methods: {
+    copy() {
+      if (isFolder(this.chosenFiles[0].type)) navigator.clipboard.writeText(`${window.location.origin}/folder?id=${this.chosenFiles[0].id}`);
+      else navigator.clipboard.writeText(`${window.location.origin}/file?id=${this.chosenFiles[0].id}`)
```
Comment (yoavrozner) [inline src/components/buttons/CopyToClipboardButton.vue:36]:
```suggestion
      const fileId = this.chosenFiles[0].id;
      if (isFolder(this.chosenFiles[0].type)) navigator.clipboard.writeText(`${window.location.origin}/folder?id=${fileId}`);
      else navigator.clipboard.writeText(`${window.location.origin}/file?id=${fileId}`)
```

File: src/components/buttons/CopyToClipboardButton.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      store.commit("onSuccess", "success.CopyToClipboard");
+    },
+    isSharedFile() {
+      const firstFile = this.chosenFiles && this.chosenFiles.length > 0 ? this.chosenFiles[0] : this.currentFolder;
+      return firstFile && firstFile.shared;
+    },
+    isSingleFile() {
+      return this.chosenFiles && this.chosenFiles.length === 1;
+    },
```
Comment (yoavrozner) [inline src/components/buttons/CopyToClipboardButton.vue:45]: These functions belongs to the utils

File: src/i18n/he.json [hunk 3]
```diff
@@ -116,7 +116,8 @@
     "DeleteShare": "ההרשאה הוסרה בהצלחה",
     "Cancel": "העלאת הקובץ בוטלה!",
     "AddFavorite": "הפריט התווסף למועדפים בהצלחה!",
-    "RemoveFavorite": "הפריט הוסר מהמועדפים בהצלחה!"
+    "RemoveFavorite": "הפריט הוסר מהמועדפים בהצלחה!",
+    "CopyToClipboard": "הקישור הועתק"
```
Comment (yoavrozner) [inline src/i18n/he.json:120]: Add it to the en.json

**Review by yoavrozner — CHANGES_REQUESTED** [review]
Well done @maya-fisher !
Please see my comments

File: public/index.html [hunk 4]
```diff
@@ -9,7 +9,7 @@
   <title>Drive</title>
 </head>
 
-<body>
+<body oncontextmenu="return false" >
```
[the remark this answers is not in the document]
Comment (maya-fisher) [reply public/index.html:12]: Yes.

[the remark this answers is not in the document]
Comment (maya-fisher) [reply public/index.html:12]: Does not work when it's in the div.

File: src/components/buttons/CopyToClipboardButton.vue [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      <v-btn
+        @click="copy()"
+        v-on="on"
+        :icon="icon"
+        :class="{ right: !icon }"
+        class="auto-margin"
+        id="move-button"
+        text
+      >
+        <img style="width: 20px; height: 20px" src="@/assets/icons/copylink.png" />
```
[the remark this answers is not in the document]
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:13]: Done.

File: src/components/buttons/CopyToClipboardButton.vue [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        <p class="button-text" v-if="!icon">{{ $t("buttons.CopyLink") }}</p>
```
[the remark this answers is not in the document]
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:14]: I have a question about this.

File: src/components/buttons/CopyToClipboardButton.vue [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    </template>
+
+    <span>{{ $t("buttons.CopyLink") }}</span>
+  </v-tooltip>
+</template>
+
+<script>
+import { mapGetters } from "vuex";
+import store from "@/store";
+import { isFolder } from "@/utils/isFolder";
```
[the remark this answers is not in the document]
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:25]: Done

File: src/components/buttons/CopyToClipboardButton.vue (hunk 1 above)
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:36]: Done.

File: src/components/buttons/CopyToClipboardButton.vue (hunk 2 above)
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:45]: Done

File: src/i18n/he.json (hunk 3 above)
Comment (maya-fisher) [reply src/i18n/he.json:120]: Done.

File: src/components/buttons/CopyToClipboardButton.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (yoavrozner) [reply src/components/buttons/CopyToClipboardButton.vue:14]: Please ask 🥇

File: src/components/buttons/CopyToClipboardButton.vue [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    <span>{{ $t("buttons.CopyLink") }}</span>
+  </v-tooltip>
+</template>
+
+<script>
+import { mapGetters } from "vuex";
+import { isFolder } from "@/utils/isFolder";
+import { isSharedFile } from "@/utils/isSharedFile";
+import { isSingleFile } from "@/utils/isSingleFile";
+import store from "@/store";
```
Comment (yoavrozner) [inline src/components/buttons/CopyToClipboardButton.vue:27]: Default import comes first

File: src/components/buttons/CopyToClipboardButton.vue [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  name: "CopyToClipboard",
+  props: ["icon"],
+  computed: {
+    ...mapGetters(["chosenFiles", "currentFolder"]),
+  },
+  methods: {
+    copy() {
+      const fileId = this.chosenFiles[0].id; 
+      if (isFolder(this.chosenFiles[0].type)) navigator.clipboard.writeText(`${window.location.origin}/folder?id=${fileId}`);
+      else navigator.clipboard.writeText(`${window.location.origin}/file?id=${fileId}`)
```
Comment (yoavrozner) [inline src/components/buttons/CopyToClipboardButton.vue:39]:
```suggestion
      const chosenFileType = this.chosenFiles[0].type;
      const url = `${window.location.origin}/${isFolder(chosenFileType) ? 'file' : 'folder'}?id=${fileId}`;
      navigator.clipboard.writeText(url)
```

File: src/utils/isSharedFile.js [hunk 10]
```diff
@@ -0,0 +1,4 @@
+export function isSharedFile(files, currentFolder) {
+    const firstFile = files && files.length > 0 ? files[0] : currentFolder;
+    return firstFile && firstFile.shared;
+}
```
Comment (yoavrozner) [inline src/utils/isSharedFile.js:4]: No newline at end of file

File: src/utils/isSingleFile.js [hunk 11]
```diff
@@ -0,0 +1,4 @@
+export function isSingleFile(files) {
+    return files && files.length === 1;
+
+}
```
Comment (yoavrozner) [inline src/utils/isSingleFile.js:4]:
```suggestion
}

```

**Review by yoavrozner — CHANGES_REQUESTED** [review]
Good job @maya-fisher !
Please see my notes

File: src/components/buttons/CopyToClipboardButton.vue (hunk 8 above)
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:27]: Done.

File: src/components/buttons/CopyToClipboardButton.vue (hunk 9 above)
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:39]: Done.

File: src/utils/isSingleFile.js (hunk 11 above)
Comment (maya-fisher) [reply src/utils/isSingleFile.js:4]: Done.

File: src/utils/isSharedFile.js (hunk 10 above)
Comment (maya-fisher) [reply src/utils/isSharedFile.js:4]: Done.

File: src/components/buttons/CopyToClipboardButton.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:14]: There is this check in almost all buttons components.
My questions is that almost all of the components of the buttons accept the icon as true, so I didn't understand what is it supposed to do. I believe that elsewhere in the ui there is a use for it, maybe you will know to tell me where exactly.

[the remark this answers is not in the document]
Comment (yoavrozner) [reply src/components/buttons/CopyToClipboardButton.vue:14]: I think we should look a this together for better insight for both of us

**Review by yoavrozner — APPROVED** [review]
@maya-fisher 🥇
Don't forget changelog.md

**Review by yarinBenisty — APPROVED** [review]
Good job @maya-fisher !

File: src/components/buttons/CopyToClipboardButton.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (maya-fisher) [reply src/components/buttons/CopyToClipboardButton.vue:14]: Great

**Pull request merged**